File: Serenity Practice/HosbitalSerenity.Web/Modules/ServerTypes/Hosbital/DoctorsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SpecialityListFormatter } from "../../Hosbital/Doctors/SpecialityListFormatter";
import { DoctorsRow } from "./DoctorsRow";

export interface DoctorsColumns {
    DoctorId: Column<DoctorsRow>;
    DoctorName: Column<DoctorsRow>;
    SpecialityList: Column<DoctorsRow>;
}

export class DoctorsColumns extends ColumnsBase<DoctorsRow> {
    static readonly columnsKey = 'Hosbital.Doctors';
    static readonly Fields = fieldsProxy<DoctorsColumns>();
}

[SpecialityListFormatter]; // referenced types

File: Serenity Practice/HosbitalSerenity.Web/Modules/Hosbital/Doctors/DoctorsColumns.cs
using Hosbital;
using Serenity.ComponentModel;
using System.ComponentModel;

namespace HosbitalSerenity.Hosbital.Columns;

[ColumnsScript("Hosbital.Doctors")]
[BasedOnRow(typeof(DoctorsRow), CheckNames = true)]
public class DoctorsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int DoctorId { get; set; }

    [EditLink]
    public string DoctorName { get; set; }

    [Width(200), SpecialityListFormatter, QuickFilter, LookupEditor(typeof(SpecialityRow), Multiple = true, InplaceAdd = false)]
    public List<int> SpecialityList { get; set; }
}